### ISSUE TITLE
Settings Sync port conflicts

### ISSUE THREAD

**Issue by PMExtra** [shanalikhan/code-settings-sync#1053]
🐛 **Describe the bug**  
Settings Sync listen port 54321 to get GitHub authorization callback. But there is a popular Chinese download tool that named `Thunder` used port 54321 too, and it will startup at login default. There are about more than hundreds million users of Thunder. So I think it would be better if we can choose another port instead 54321.

📺 **Error phenomenon** 
The browser show `ERR_EMPTY_RESPONSE` when I clicked `LOGIN WITH GITHUB` button on `Welcome to Settings Sync` page.

📺  **Additional context**
![TIM截图20190928165210](https://user-images.githubusercontent.com/11289158/65814108-5a121780-e210-11e9-82cd-72916201e5b1.png)

**Comment by shanalikhan** [comment]
Alright, i think we can make this port as a configuration so anyone can change when required

**Comment by PMExtra** [comment]
I think we can make it random and pre-check if the random port is already in use.

**Comment by shanalikhan** [comment]
Yes, that is good too. Will look into it or PR is welcome.

https://github.com/shanalikhan/code-settings-sync/blob/eb332ba5e8180680e613e94be89119119c5638d1/src/service/webview.service.ts#L348

**Comment by auxves** [comment]
@shanalikhan This won't work since the port must be set in the GitHub OAuth application settings and cannot be dynamic.

**Comment by shanalikhan** [comment]
Oh! I forgot that. You are right. 
Do u have any ideas in mind?

That means we really cant change the port in new version, there will be backward compatibility issues - Settings Sync GUI will stop working for all of the previous versions.

Lets see how this thread go, i will keep this opened
Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by auxves** [comment]
> Do u have any ideas in mind?

One solution could be to create a new GitHub OAuth application based on the current one, change the port to something very uncommon on both the OAuth app and Settings Sync, and use that application's client ID, secret, and port in the new version. That way, older versions will still work, but new versions will use the new port.

**Comment by shanalikhan** [comment]
Yes, but in that case people are going to authorizing another Settings Sync application to their GitHub profile...

**Comment by PMExtra** [comment]
At least, we should provide a message to notify users the port was occupied, and link to a wiki to guide users how to check which program listened the port, or how to configure manually.

**Comment by shanalikhan** [comment]
> we should provide a message to notify users the port was occupied, and link to a wiki to guide users how to check which program listened the port, or how to configure manually.

Public Edit is allowed on [Wiki Troubleshooting](https://github.com/shanalikhan/code-settings-sync/wiki/Troubleshooting). Feel free to improve and help community. Thanks

**Comment by xiangnanscu** [comment]
thanks dude. you save my day

**Comment by fsqx** [comment]
thank you so much for save me.
